services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/CrtExceptionTransformationIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.transfer.s3.internal;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static software.amazon.awssdk.testutils.service.S3BucketUtils.temporaryBucketName;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.testutils.RandomTempFile;
+import software.amazon.awssdk.transfer.s3.DownloadRequest;
+import software.amazon.awssdk.transfer.s3.S3IntegrationTestBase;
+import software.amazon.awssdk.transfer.s3.S3TransferManager;
+import software.amazon.awssdk.transfer.s3.UploadRequest;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class CrtExceptionTransformationIntegrationTest extends S3IntegrationTestBase {
+
+    private static final String BUCKET = temporaryBucketName(CrtExceptionTransformationIntegrationTest.class);
+
+    private static final String KEY = "some-key";
+
+    private static final int OBJ_SIZE = 8 * 1024;
+    private static RandomTempFile testFile;
+    private S3TransferManager transferManager;
+    private S3CrtAsyncClient s3Crt;
+
+    @BeforeClass
+    public static void setupFixture() {
+        createBucket(BUCKET);
+    }
+
+    @AfterClass
+    public static void tearDownFixture() {
+        deleteBucketAndAllContents(BUCKET);
+    }
+
+    @Before
+    public void methodSetup() throws IOException {
+        testFile = new RandomTempFile(BUCKET, OBJ_SIZE);
+        s3Crt = S3CrtAsyncClient.builder()
+                .credentialsProvider(CREDENTIALS_PROVIDER_CHAIN)
+                .region(S3IntegrationTestBase.DEFAULT_REGION)
+                .build();
+        transferManager =
+                S3TransferManager.builder()
+                        .s3ClientConfiguration(b -> b.credentialsProvider(CREDENTIALS_PROVIDER_CHAIN)
+                                .region(S3IntegrationTestBase.DEFAULT_REGION)
+                                .targetThroughputInGbps(20.0)
+                                .minimumPartSizeInBytes(1000L))
+                        .build();
+    }
+
+    @Test
+    public void getObjectNoSuchKey() throws IOException {
+        String randomBaseDirectory = Files.createTempDirectory(getClass().getSimpleName()).toString();
+        assertThatThrownBy(() -> s3Crt.getObject(GetObjectRequest.builder().bucket(BUCKET).key("randomKey").build(),
+                Paths.get(randomBaseDirectory).resolve("testFile")).get())
+                .hasCauseInstanceOf(SdkServiceException.class)
+                .hasMessageContaining("Failed with Aws Error code \"NoSuchKey\" and Aws Error Message \"The specified key does not exist.\"");
```

Comment:
Minor: we probably don't need to verify the whole message because services can change them, which will make the tests flaky.

Same as other tests

Reply:
removed this